### PR TITLE
feat(charts): Add Prometheus chart demo

### LINKED ISSUE #5996: Create a demo to consume Prometheus data in PatternFly chart
**Describe the issue. What is the expected and unexpected behavior?**
We need a demo to show how to consume data from the Prometheus database and render that data using the PatternFly chart components.

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
Enhancement.

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
**What**: Closes #5996

### CODE REVIEW

**Comment by wojta** [comment]
@DaoDaoNoCode Is it meant to be with that big question mark? 
image
However the proper demo is available in a new window by clicking the question mark.

File: packages/react-core/src/demos/Chart/PrometheusChart.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+                {data.map((series, i) => {
+                  const [metric, values] = series;
+                  if (values === null) {
+                    return null;
+                  }
+                  const color = colors[i % colors.length];
+                  const style = {
+                    data: { [isStacked ? 'fill' : 'stroke']: color },
+                    labels: { fill: color, name: tooltipSeriesNames[i] }
+                  };
```
Comment (dlabrecq) [inline packages/react-core/src/demos/Chart/PrometheusChart.md:194]: You shouldn't need to define the color style, unless overriding colors. The chart will iterate over the color scale defined by the multiUnordered theme.

File: packages/react-core/src/demos/Chart/PrometheusChart.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        <ChartLegendTooltip legendData={legendData} title={datum => dateTimeFormatterWithSeconds.format(datum.x)} />
+      }
+      mouseFollowTooltips
+      voronoiDimension="x"
+      voronoiPadding={50}
+    />
+  );
+
+  const theme = getCustomTheme(ChartThemeColor.multiUnordered, ChartThemeVariant.light);
+  const colors = theme.line.colorScale;
```
Comment (dlabrecq) [inline packages/react-core/src/demos/Chart/PrometheusChart.md:130]: Unless you are changing the theme, you don't need this. You can simply define the chart's color scale using:

`themeColor={ChartThemeColor.multiUnordered}`

File: packages/react-core/src/demos/Chart/PrometheusChart.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+                  if (values === null) {
+                    return null;
+                  }
+                  const color = colors[i % colors.length];
+                  const style = {
+                    data: { [isStacked ? 'fill' : 'stroke']: color },
+                    labels: { fill: color, name: tooltipSeriesNames[i] }
+                  };
+                  return (
+                    <ChartComponent data={values} groupComponent={<g />} key={i} name={metric.cpu} style={style} />
```
Comment (dlabrecq) [inline packages/react-core/src/demos/Chart/PrometheusChart.md:196]: Is there a reason to override the `groupComponent`?

File: packages/react-core/src/demos/Chart/PrometheusChart.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+```js isFullscreen
+import React from 'react';
+import {
+  Avatar,
+  Brand,
+  Checkbox,
+  Divider,
+  Nav,
+  NavItem,
+  NavList,
```
Comment (dlabrecq) [inline packages/react-core/src/demos/Chart/PrometheusChart.md:24]: These components don't appear to be in use

File: packages/react-core/src/demos/Chart/PrometheusChart.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+  const ChartComponent = isStacked ? ChartArea : ChartLine;
+
+  const PageToolbar = <div>need to implement new toolbar</div>;
+  const Header = (
+    <PageHeader
+      logo={<Brand src={imgBrand} alt="Patternfly Logo" />}
+      toolbar={PageToolbar}
+      avatar={<Avatar src={imgAvatar} alt="Avatar image" />}
+    />
+  );
```
Comment (dlabrecq) [inline packages/react-core/src/demos/Chart/PrometheusChart.md:146]: Not sure if we need to include the PatternFly avatar in the page header? The other demos don't seem to use it?

**Review by dlabrecq — CHANGES_REQUESTED** [review]
FYI, codesandbox.io couldn't load the mock data.

**Comment by nicolethoen** [comment]
@DaoDaoNoCode To generate the snapshots so that the big question mark goes away:

yarn build:docs && yarn serve:docs
(new terminal) yarn workspace @patternfly/react-docs screenshots (you can add "screenshots": "yarn workspace @patternfly/react-docs screenshots" to the root package.json to just have to run yarn screenshots in the future)

Let me know if you have questions

**Comment by DaoDaoNoCode** [comment]
@wojta @dlabrecq After checking with Jan, this demo might not be the solution we are looking for. We actually want a UXD extension like the QuickStart, to easily use the Prometheus chart across different managed applications and keep the styles aligned. More details to be discussed.
And thanks for the hint @nicolethoen ! BTW, Would you like to give us a brief introduction about the UXD extension as you mentioned this morning in the team meeting after we decide the next step?
I will close this PR for now, thanks 🙂

**Pull request closed without merging**